src/amountToHangul.ts

```diff
@@ -26,10 +26,12 @@ export const HANGUL_NUMBERS_FOR_DECIMAL = ['영', '일', '이', '삼', '사', '
 export const HANGUL_CARDINAL = ['', '십', '백', '천'];
 
 export function amountToHangul(amount: string | number) {
-  const [integerPart, tempDecimalPart] = String(amount)
+  const [tempIntegerPart, tempDecimalPart] = String(amount)
     .replace(/[^\d.]+/g, '')
     .split('.');
 
+  const integerPart = tempIntegerPart !== '0' ? tempIntegerPart.replace(/^0+/, '') : tempIntegerPart;
```

Comment:
요렇게 temp가 아닌 조금 더 명확한 이름을 사용해보면 어떤가요?
또한 decimal같은경우는 temp가 붙을 필요가 없을것 같기두 해요!

```suggestion
  const [rawIntegerPart, decimalPart] = String(amount)
    .replace(/[^\d.]+/g, '')
    .split('.');

  const integerPart = rawIntegerPart !== '0' ? rawIntegerPart.replace(/^0+/, '') : rawIntegerPart;
```

Reply:
> 요렇게 temp가 아닌 조금 더 명확한 이름을 사용해보면 어떤가요? 또한 decimal같은경우는 temp가 붙을 필요가 없을것 같기두 해요!

훨씬 좋은 것 같아요!!

demicalPart의 경우에도 소수부 뒷자리의 0을 제거하는 정규표현식을 한 번 거치는 과정이 있더라구요!

```ts
const decimalPart = tempDecimalPart?.replace(/0+$/, '');
```

이 부분도 tempDemicalPart 대신 rawDemicalPart로 네이밍하는 게 더 좋을 것 같네요 ☺️